src/components/ConcentratedLiquidity/DetailModal/Tabs/DetailTab/index.tsx

```diff
@@ -1,79 +1,181 @@
+import { DetailTabProps } from './types';
 
-const DetailTab: React.FC = () => {
+const DetailTab: React.FC<DetailTabProps> = (props) => {
+  const { position } = props;
+  const theme = useContext(ThemeContext);
   const { t } = useTranslation();
-  // ------------------------------ MOCK DATA ----------------------------------
-  const currency0 = new Token(ChainId.AVALANCHE, '0xb97ef9ef8734c71904d8002f8b6bc66dd9c48a6e', 6, 'USDC', 'USD Coin');
-  const currency1 = new Token(
-    ChainId.AVALANCHE,
-    CHAINS[ChainId.AVALANCHE].contracts!.png,
-    18,
-    CHAINS[ChainId.AVALANCHE].png_symbol!,
-    'Pangolin',
-  );
+  const chainId = useChainId();
+  const useUnderlyingTokens = useUnderlyingTokensHook[chainId];
+  const usePositionTokenURI = usePositionTokenURIHook[chainId];
+  const metadata = usePositionTokenURI(position?.tokenId);
+  const [, pool] = usePool(position?.token0 ?? undefined, position?.token1 ?? undefined, position?.fee);
+  const nativePosition = useMemo(() => {
+    if (
+      pool &&
+      position?.liquidity &&
+      typeof position?.tickLower === 'number' &&
+      typeof position?.tickUpper === 'number'
+    ) {
+      return new Position({
+        pool,
+        liquidity: position?.liquidity.toString(),
+        tickLower: position?.tickLower,
+        tickUpper: position?.tickUpper,
+      });
+    }
+    return undefined;
+  }, [position?.liquidity, pool, position?.tickLower, position?.tickUpper]);
+
+  const tickAtLimit = useIsTickAtLimit(position?.fee, position?.tickLower, position?.tickUpper);
+  const pricesFromPosition = getPriceOrderingFromPositionForUI(nativePosition);
+  const { priceLower, priceUpper, base } = useInverter({
+    priceLower: pricesFromPosition.priceLower,
+    priceUpper: pricesFromPosition.priceUpper,
+    quote: pricesFromPosition.quote,
+    base: pricesFromPosition.base,
+    invert: false, // we don't want to invert the price, just the tokens
+  });
+  const inverted = position?.token1 ? base?.equals(position?.token1) : undefined;
+  const currentPrice = pool && formatPrice(inverted ? pool?.token1Price : pool?.token0Price, NumberType.TokenTx);
+  const minPrice = formatTickPrice({
+    price: priceLower,
+    atLimit: tickAtLimit,
+    direction: Bound.LOWER,
+  });
+  const maxPrice = formatTickPrice({
+    price: priceUpper,
+    atLimit: tickAtLimit,
+    direction: Bound.UPPER,
+  });
+
+  // TODO: Open it in production
+  // -------------------------------------------------------
+  // const useUSDCPrice = useUSDCPriceHook[chainId];
+  // const price0 = useUSDCPrice(position?.token0 ?? undefined);
+  // const price1 = useUSDCPrice(position?.token1 ?? undefined);
+  // -------------------------------------------------------
+  const USDC = USDCe[chainId];
+  const price0 = position?.token0 && new Price(position.token0, USDC, '1', '1');
+  const price1 = position?.token1 && new Price(position.token1, USDC, '1', '1');
+  // -------------------------------------------------------
```

Comment:
should we fix this now ? is `useUSDCPrice` hook creating issue in fuji ?

Reply:
I've opened it.

Reply:
yes thats fine but atleast we have code in place so we can use test it in mainnet